src/QsCompiler/Core/TransformationDefinition.fs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Transformations.Core
+    
+
+type QsSyntaxTreeTransformation<'T> private (state : 'T, dummy) =     
```

Comment:
Is dummy needed here? It seems like it is unused.

Reply:
I haven't found a more elegant way of dealing with this yet - hopefully I'll have a good inspiration at some point. Basically F# is not happy to have two constructors with the same signature (for obvious reasons), but I need a) a base constructor that allows me to build a QsSyntaxTreeTransformation in the first place such that I can get a handle to it to construct the subtransformations, and b) an additional constructor that actually initializes the subtransformations with the ones that have the correct handle back to the parent. 
So yeah, not exactly the prettiest way to achieve this, but also the only one that came to mind.